lib/ansible/modules/windows/win_regedit.ps1

```diff
@@ -235,59 +382,115 @@ if ($type -in @("binary", "none")) {
     }
 }
 
-
 # convert the type string to the .NET class
 $type = [System.Enum]::Parse([Microsoft.Win32.RegistryValueKind], $type, $true)
 
-if ($state -eq "present") {
-    if (-not (Test-Path -path $path)) {
-        # the key doesn't exist, create it so the next steps work
-        try {
-            New-Item -Path $path -Type directory -Force -WhatIf:$check_mode
-        } catch {
-            Fail-Json $result "failed to create registry key at $($path): $($_.Exception.Message)"
-        }
-        $result.changed = $true
+if ($hive) {
+    if (-not (Test-Path $hive)) {
+        Fail-Json -obj $result -message "hive at path '$hive' is not valid or accessible, cannot load hive"
+    }
+    if (Test-Path -Path HKLM:\ANSIBLE) {
+        Fail-Json -obj $result -message "hive already loaded at HKLM:\ANSIBLE, cannot load another hive until this is manually unloaded"
```

Comment:
This to me is problematic. We can end up in a situation where the only way to get out of this is to do something outside of Ansible. Ideally it would only be loaded within its own context, and automatically unloaded.

I wonder if we could either make this path unique to the session, and have a clean-up function for older sessions.

Reply:
We could make the path unique but would need more complex handling around how someone would set the path. This isn’t too much of an issue as keys can’t be created at the root folder and this would only fire if a hive has been manually loaded before.

I’ll let others comment on this before changing though as I traded convenience for simpler code in this case.

Reply:
Hey @dagwieers, I've updated the code so that it will unload the hive if it has been set and output a warning if this happens.

Reply:
That's very useful, but doesn't mitigate the possibility of conflicts, so maybe a better implementation (in time) is warranted. Maybe adding a note in the code wrt. this.

Is the unloading of a hive guaranteed to work, or could it fail when in use ?

Reply:
We will want to make the decision before merging this in as it would affect what the user would put in their `path`. In the current case the hive is accessed with `HKLM:\ANSIBLE\*` while in your suggestion it would come under its own root. I'm not even sure if you can create a Registry PSDrive and map it to a subkey on an existing drive.

As with all things it can fail, the main reason would be if someone had loaded the hive manually before hand and still had a handle open to it. Even if we were to use unique names you cannot load a hive file again once it has been loaded once under a different name.

Reply:
Ouch, if it can only be loaded once... This is fine by me.